File: leaf-system/src/main/java/com/leaf/domain/LeafDept.java
package com.leaf.domain;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * @author jhlz
 * @version 1.0.0
 */
@Entity
@Table(name = "leaf_dept")
public interface LeafDept extends BaseEntity, ID {

    /**
     * 部门名称
     */
    @Key
    String deptName();

    /**
     * 父级部门 ID
     */
    Long parentId();

    /**
     * 部门领导
     */
    long leaderId();

    /**
     * 部门描述
     */
    @Nullable
    String description();

    /**
     * 状态：0正常，1禁用
     */
    int status();

    /**
     * 排序
     */
    int sortable();

    /**
     * 父级列表
     */
    @Nullable
    String ancestors();

    @OneToMany(mappedBy = "dept")
    List<LeafUser> userList();
}

File: src/main/kotlin/com/jhlz/domain/LeafDept.kt
package com.jhlz.domain

import org.babyfish.jimmer.sql.*

/**
 * @author jhlz
 * @version 1.0.0
 */
@Entity
@Table(name = "leaf_dept")
interface LeafDept : BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    /**
     * 部门名称
     */
    @Key
    val deptName: String

    /**
     * 父部门 id
     */
    val parentId: Long?

    /**
     * 父级列表
     */
    val ancestors: String?

    /**
     * 部门领导 id
     */
    val leaderId: Long?

    /**
     * 状态：0正常，1禁用
     */
    val status: Int

    /**
     * 排序
     */
    val sortable: Int

    @OneToMany(mappedBy = "dept")
    val users: List<LeafUser>
}
